crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -69,6 +103,8 @@ pub enum WellKnownNote {
     P2ID,
     P2IDE,
     SWAP,
+    MINT,
+    BURN,
```

Comment:
I wonder if `MINT` and `BURN` are a bit too generic. Currently, they are specifically for minting and burning of fungible assets. I think we can take two routes we can take here:

1. Keep naming as is, and in the future extend these to also cover non-fungible assets. For `BURN` note that should be relatively easy - not sure about the `MINT` note.
2. Change the naming to be fungible asset specific - e.g., `MINTFA` and `BURNFA`. Or maybe the naming convention should reflect that these notes are specifically for the network fungible faucet. Then it could be `MINT4NFF` ("mint-for-network-fungible-faucet").

I think my preference would be to go with the first approach for now.

Reply:
Agreed, I think most use cases we've seen so far from pioneers are using fungible assets, so let's support NFTs when the demand is there (I think there'd be generally less demand for original NFT use cases on a private chain, but there's probably a bunch of use cases that can be ported from Ethereum)